src/history/history.go

```diff
@@ -114,7 +117,13 @@ func SetState(repoURL string, cur State) error {
 	cur.Timestamp = time.Now()
 	var err error
 	if cur.Status != "deploying" || checkSubdomain(cur.Subdomain) {
```

Comment:
In go it is pretty common to keep the flow of control easy without a bunch of nested ifs. See https://golang.org/doc/effective_go.html#if
Do something like:
```
if cur.Status == "deploying" && subdomainUsed(cur.Subdomain) {
   return fmt.Errorf("subdomain in use ...")
}
if providedTag != oldTag {
   return fmt.Errorf("provided tag does not match original tag, something changed in between ..."
}
history[repoUrl].Current = &cur
// compute the new tag and set it.
```

Reply:
I've changed it to remove nested if's in the latest commit, but couldn't remove the else's since another function had to be run before returning.